src/quantization/scalar_quantization_trainer.cpp

```diff
@@ -74,11 +76,45 @@ ScalarQuantizationTrainer::classic_train(const float* data,
     }
 }
 
+void
+ScalarQuantizationTrainer::trunc_bound_train(const float* data,
+                                             uint64_t count,
+                                             float* upper_bound,
+                                             float* lower_bound) const {
+    auto ignore_count = static_cast<uint64_t>(static_cast<float>(count - 1) * 0.001);
+
+    for (uint64_t i = 0; i < dim_; ++i) {
+        upper_bound[i] = std::numeric_limits<float>::lowest();
+        lower_bound[i] = std::numeric_limits<float>::max();
+        std::priority_queue<float, std::vector<float>, std::greater<>> heap_max;
+        std::priority_queue<float, std::vector<float>, std::less<>> heap_min;
+        heap_max.emplace(upper_bound[i]);
+        heap_min.emplace(lower_bound[i]);
+        for (uint64_t j = 0; j < count; ++j) {
+            auto value = data[j * dim_ + i];
+            if (value > heap_max.top() || heap_max.size() < ignore_count) {
```

Comment:
heap_max.size() < ignore_count || value > heap_max.top()

Reply:
why ? heap_max.size() < ignore_count is more stringent